### PR TITLE
Feature/migration to newer version

### PR DESCRIPTION
**Pull request by Oliweirdo**
Currently we are using .NET Core 2.2 which has reached end of support on 23rd december 2019. We should move to current LTS version.

## Motivation and context

#273
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/DevAdventCalendarCompetition.csproj [hunk 1]
```diff
@@ -24,16 +25,17 @@
     <PackageReference Include="Microsoft.AspNetCore.Authentication.Twitter" Version="2.2.0" />
     <PackageReference Include="Microsoft.AspNetCore.CookiePolicy" Version="2.2.0" />
     <PackageReference Include="Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore" Version="2.2.1" />
+    <PackageReference Include="Microsoft.AspNetCore.Http.Abstractions" Version="2.2.0" />
     <PackageReference Include="Microsoft.AspNetCore.HttpOverrides" Version="2.2.0" />
     <PackageReference Include="Microsoft.AspNetCore.HttpsPolicy" Version="2.2.0" />
-      <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.2.0" />
+      <!--<PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.2.0" />-->
```
Comment (WTobor) [inline src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/DevAdventCalendarCompetition.csproj]: why there is a comment?

File: src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/DevAdventCalendarCompetition.csproj [hunk 2]
```diff
@@ -24,16 +25,17 @@
...
     <PackageReference Include="Microsoft.AspNetCore.Mvc.ViewFeatures" Version="2.2.0" />
     <PackageReference Include="Microsoft.AspNetCore.Owin" Version="2.2.0" />
     <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.9.4">
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
-    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="2.2.0" />
+    <!--<PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="2.2.0" />-->
```
Comment (WTobor) [inline src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/DevAdventCalendarCompetition.csproj]: why there is a comment?

**Comment by WTobor** [comment]
everything looks great, but I cannot log in - @Oliweirdo could you check that?

**Comment by smiechuskce** [comment]
> everything looks great, but I cannot log in - @Oliweirdo could you check that? Maybe there is a problem with routing?

image

@Oliweirdo This specific issue was possibly caused by the breaking changes of EF 3.0 in executing LINQ queries:

https://github.com/dotnet/efcore/issues/17068

And it is related to https://docs.microsoft.com/en-us/ef/core/what-is-new/ef-core-3.0/breaking-changes#linq-queries-are-no-longer-evaluated-on-the-client

So I think any code using GroupBy might cause the exception above, so all of our repositories codebase should be reviewed.

**Comment by WTobor** [comment]
@Oliweirdo - I have fixed the configuration, user can log in
@smiechuskce - I have fixed the problem with _GroupBy()_ - thanks :)

@patrikulus please review if we still need additional configuration ex. _StartupExtention.cs_, _UseHttpsRequestScheme()_ method etc. :)

**Pull request merged**